src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+  // static alive() {
+  //   return this.aliveAnimals;
+  // }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herb) {
+    if (herb.hidden === false) {
+      herb.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
In the `bite` method, it would be prudent to check if the `herb` parameter is an instance of `Herbivore` before attempting to modify its `health` property. This can prevent potential errors if a non-Herbivore object is passed to this method.